packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -482,6 +482,404 @@ describe( 'Default search suggestions', () => {
+						container
+					);
+				} );
+				// Await the initial suggestions to be fetched
+				await eventLoopTick();
+
+				// Search Input UI
+				const searchInput = container.querySelector(
+					'input[aria-label="URL"]'
+				);
+
+				// TODO: select these by aria relationship to autocomplete rather than arbitary selector.
+				const searchResultElements = container.querySelectorAll(
+					'[role="listbox"] [role="option"]'
+				);
+				const createButton = first(
+					Array.from( searchResultElements ).filter( ( result ) =>
+						result.innerHTML.includes( 'Create new' )
+					)
+				);
+
+				// Verify input has no value
+				expect( searchInput.value ).toBe( '' );
+				expect( createButton ).toBeFalsy(); // shouldn't exist!
+			}
+		);
+
+		it( 'should not show not show an option to create an entity when input is empty', async () => {
+			act( () => {
+				render(
+					<LinkControl
+						showInitialSuggestions={ true } // should show even if we're not showing initial suggestions
+						showCreateEntity={ true }
+						createEntity={ jest.fn() }
+					/>,
+					container
+				);
+			} );
+			// Await the initial suggestions to be fetched
+			await eventLoopTick();
+
+			// Search Input UI
+			const searchInput = container.querySelector(
+				'input[aria-label="URL"]'
+			);
+
+			// TODO: select these by aria relationship to autocomplete rather than arbitary selector.
+			const searchResultElements = container.querySelectorAll(
+				'[role="listbox"] [role="option"]'
+			);
+			const createButton = first(
+				Array.from( searchResultElements ).filter( ( result ) =>
+					result.innerHTML.includes( 'Create new' )
+				)
+			);
+
+			// Verify input has no value
+			expect( searchInput.value ).toBe( '' );
+			expect( createButton ).toBeFalsy(); // shouldn't exist!
+		} );
+
+		// it.each( [
+		// 	'https://wordpress.org',
+		// 	'www.wordpress.org',
+		// 	'mailto:example123456@wordpress.org',
+		// 	'tel:example123456@wordpress.org',
+		// 	'#internal-anchor',
+		// ] )( 'should not show option to "Create Page" when text is a form of direct entry (eg: %s)', async ( inputText ) => {
+		// 	act( () => {
+		// 		render(
+		// 			<LinkControl
+		// 				showCreateEntity={ true }
+		// 				createEntity={ jest.fn() }
+		// 			/>, container
+		// 		);
+		// 	} );
+
+		// 	// Search Input UI
+		// 	const searchInput = container.querySelector( 'input[aria-label="URL"]' );
+
+		// 	// Simulate searching for a term
+		// 	act( () => {
+		// 		Simulate.change( searchInput, { target: { value: inputText } } );
+		// 	} );
+
+		// 	await eventLoopTick();
+
+		// 	// TODO: select these by aria relationship to autocomplete rather than arbitary selector.
+		// 	const searchResultElements = container.querySelectorAll( '[role="listbox"] [role="option"]' );
+
+		// 	const createButton = Array.from( searchResultElements ).filter( ( result ) => result.innerHTML.includes( 'Create new' ) );
+
+		// 	expect( createButton ).toBeNull();
+		// } );
```

Comment:
Commented code should not be committed.

Reply:
Agreed. I shouldn't have done this. It was causing trouble when testing as these will not pass until your update to `@wordpress/url` is merged.

I've reinstated as a guard now.

Reply:
I've rebased to include the updates merged in https://github.com/WordPress/gutenberg/pull/19871